tools/spectral/ipa/rulesets/functions/utils/compareUtils.js

```diff
@@ -72,19 +99,46 @@ export function removePropertyByFlag(schema, propertyToRemove) {
 }
 
 /**
- * Recursively removes properties with readOnly: true flag from schema
+ * Recursively removes specific property keys from schema
+ * @param {object} schema The schema to process
+ * @param {...string} propertyNames Property names to remove
+ * @returns {object} New schema with specified property keys removed
+ */
+export function removePropertyKeys(schema, ...propertyNames) {
+  if (!schema || typeof schema !== 'object') return schema;
+
+  const result = {};
+
+  // Handle regular object properties
+  for (const [key, value] of Object.entries(schema)) {
+    // Skip this property if it's in the list of properties to remove
+    if (propertyNames.includes(key)) continue;
+
+    if (typeof value === 'object') {
+      result[key] = removePropertyKeys(value, ...propertyNames);
+    } else {
+      result[key] = value;
+    }
+  }
+
+  return result;
+}
+/**
+ * Recursively removes properties for Response schemas
  * @param {object} schema The schema to process
  * @returns {object} New schema with readOnly properties removed
  */
-export function removeReadOnlyProperties(schema) {
-  return removePropertyByFlag(schema, 'readOnly');
+export function removeResponseProperties(schema) {
+  let result = removePropertiesByFlag(schema, 'readOnly');
+  return removePropertyKeys(result, 'title', 'description', 'required');
 }
 
 /**
- * Recursively removes properties with writeOnly: true flag from schema
+ * Recursively removes properties for Request schemas
  * @param {object} schema The schema to process
  * @returns {object} New schema with writeOnly properties removed
  */
-export function removeWriteOnlyProperties(schema) {
-  return removePropertyByFlag(schema, 'writeOnly');
+export function removeRequestProperties(schema) {
+  let result = removePropertiesByFlag(schema, 'writeOnly');
+  return removePropertyKeys(result, 'title', 'description', 'required');
```

Comment:
hardcoding those values in generic helper method is technical debt. Helper should get those as array. 
Otherwise we might be seeing ourselves creating urgent PR's to add more properties etc. over the time.